source/renderer/app/components/wallet/WalletSendForm.js

```diff
@@ -378,48 +378,55 @@ export default class WalletSendForm extends Component<Props, State> {
     }
   }
 
-  async _calculateTransactionFee(address: string, amountValue: string) {
+  calculateTransactionFee = async (address: string, amountValue: string) => {
     const amount = formattedAmountToLovelace(amountValue);
-    this.setState({
-      isCalculatingTransactionFee: true,
+    const {
+      feeCalculationRequestQue: prevFeeCalculationRequestQue,
+    } = this.state;
+    this.setState((prevState) => ({
       isTransactionFeeCalculated: false,
       transactionFee: new BigNumber(0),
       transactionFeeError: null,
-    });
+      feeCalculationRequestQue: prevState.feeCalculationRequestQue + 1,
+    }));
     try {
       const fee = await this.props.calculateTransactionFee(address, amount);
-      if (this._isMounted) {
-        this._isCalculatingFee = false;
+      if (
+        this._isMounted &&
+        this.state.feeCalculationRequestQue - prevFeeCalculationRequestQue === 1
```

Comment:
@yakovkaravelov can we turn this line into a variable? That way it will be easier to understand what it means and we can reuse it on the 408 line...